Source/Globals.cpp

```diff
@@ -59,7 +61,23 @@ namespace android::global
     void Initialize(JavaVM* javaVM, jobject context)
     {
         g_javaVM = javaVM;
-        g_appContext = GetEnvForCurrentThread()->NewGlobalRef(android::content::Context{context}.getApplicationContext());
+        JNIEnv* env{GetEnvForCurrentThread()};
+        g_appContext = env->NewGlobalRef(android::content::Context{context}.getApplicationContext());
+        // initialize java websocket class here and cache it
+        jclass webSocketClass = env->FindClass("com/jsruntimehost/unittests/WebSocket");
+        {
+            if (!webSocketClass) {
+                env->ExceptionClear();
+                std::cerr << "Java Websocket class not found." << std::endl;
+            } else {
+                g_webSocketClass = (jclass) env->NewGlobalRef(webSocketClass);
```

Comment:
Are you disposing this Global reference somewhere?

Reply:
Have refatored the initializtion of the java websocket class into JavaWrappers.cpp and added the method to delete the global reference.